pylint/checkers/base/basic_checker.py

```diff
@@ -657,13 +657,38 @@ def _check_misplaced_format_function(self, call_node: nodes.Call) -> None:
             ):
                 self.add_message("misplaced-format-function", node=call_node)
 
+    @staticmethod
+    def _is_sys_exit(node: nodes.Call) -> bool:
+        """Detect call to sys.exit."""
+        if not isinstance(node.func, nodes.Attribute) or isinstance(
+            node.parent, nodes.Lambda
+        ):
+            return False
+
+        try:
+            inferred = next(node.func.infer())
+        except astroid.InferenceError:
+            return False
+
+        return inferred.qname() == "sys.exit"
+
     @utils.only_required_for_messages(
-        "eval-used", "exec-used", "bad-reversed-sequence", "misplaced-format-function"
+        "eval-used",
+        "exec-used",
+        "bad-reversed-sequence",
+        "misplaced-format-function",
+        "unreachable",
     )
     def visit_call(self, node: nodes.Call) -> None:
-        """Visit a Call node -> check if this is not a disallowed builtin
-        call and check for * or ** use.
+        """Visit a Call node ->
+        1
+
+        check if this is not a disallowed builtin call
+        2. check for * or ** use.
+        3. check if this call is sys.exit and is followed by unreachable code
         """
+        if self._is_sys_exit(node):
+            self._check_unreachable(node)
```

Comment:
The confidence need to be INFERENCE in this case, but it should still be HIGH when we don't need inference.